src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -132,11 +133,18 @@ export const NavSearchBar: React.FC<NavSearchBarProps> = ({ venueId }) => {
   const { worldUsers } = useWorldUsers();
   const { openUserProfileModal } = useProfileModalControls();
 
-  const foundUsers = useMemo<JSX.Element[]>(() => {
-    if (!searchQuery) return [];
+  const { foundUsers, foundUsersLength } = useMemo<{
```

Comment:
I'd rather extract rendering of users to another component and have the JSX done there without a memo, but i wanted to limit the scope of the change.
e.g. 

```typescript
export const NavSearchUserResults: React.FC<{ users: WithId<User>[], onClick: () => {} }> = ({ users, onClick }) => {
  return (
    <>
      {users.map((user) => (
        <NavSearchResult
          key={`user-${user.id}`}
          title={user.partyName ?? DEFAULT_PARTY_NAME}
          user={user}
          onClick={() => onClick(0)}
        />
      ))}
    </>
  )
}

```